odoo/addons/base/tests/test_qweb.py

```diff
@@ -1286,6 +1286,124 @@ def test_error_message_4(self):
         except AssertionError as e:
             self.assertIn('template is required', str(e))
 
+    def test_error_message_5(self):
+        # Error not found a first rendering.
+        t = self.env['ir.ui.view'].create({
+            'name': 'test',
+            'type': 'qweb',
+            'arch_db': '''<t t-name="test">
+                <section>
+                    <div t-esc="abc + def + (">
+                        <span>content</span>
+                    </div>
+                </section>
+            </t>'''
+        })
+
+        def load(ref_alias):
+            raise ValueError(f'Not Found: {ref_alias}')
+
+        with self.assertRaises(ValueError):
+            self.env['ir.qweb']._render(t.id, load=load)
+
+        try:
+            self.env['ir.qweb']._render(t.id, load=load)
+        except ValueError as e:
+            self.assertEqual(f'Not Found: {t.id}', str(e))
```

Comment:
Your code doesn't verify there is an exception. You should add an `else` clause containing `assertTrue(False)` or (better) use `assertRaises` as follow:
```suggestion
        with self.assertRaises(ValueError, msg=f"Not Found: {t.id}"):
            self.env['ir.qweb']._render(t.id, load=load)
```
See: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises